uk.co.compucorp.civicrm.hrcontactactionsmenu/tests/phpunit/CRM/HRContactActionsMenu/Hook/AddContactMenuActionsTest.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+use CRM_HRContactActionsMenu_Hook_AddContactMenuActions as AddContactMenuActionsHook;
+use CRM_HRContactActionsMenu_Component_Menu as ActionsMenu;
+
+/**
+ * Class CRM_HRContactActionsMenu_Hook_AddContactMenuActionsTest
+ *
+ * @group headless
+ */
+class CRM_HRContactActionsMenu_Hook_AddContactMenuActionsTest extends BaseHeadlessTest {
+
+  public function testInvokeReturnsAnActionsMenuInstance() {
+    $this->assertInstanceOf(ActionsMenu::class, AddContactMenuActionsHook::invoke());
+  }
```

Comment:
It would also be interesting to add a test to check that the invoke() method actually triggers the hook. You can use the `HookInterface` to help you with that. Check the example in the interface's docblock

Reply:
I tried that and took a look at some core tests that tested hooks, Most of them call an API or method that is supposed to implement/call the particular hook and then define the hook implementation in the test class. However in this particular case, The hook is not yet implemented or called by other extensions so I couldn't go with that approach.

Reply:
I'm not sure I understand. How come is the hook is not called? You can just call it by calling the `invoke()`, can't you?

For the test, I was thinking of something like this:

```php
// This will be called when AddContactMenuActions::invoke() invokes the hook
public function hook_civicrm_addContactMenuActions(ActionsMenu $menu) {
  $dummyGroup = new ActionsGroup('FooBar');
  $menu->addToMainPanelGroup($dummyGroup);
}

public function testInvokeInvokesTheAddContactMenuActionsHook() {
  // Invokes the hook, which will call the hook implementation in this class, which will add a 
 // group to the menu
   $menu = AddContactMenuActions::invoke();
   $groups = $menu->getMainPanelItems();
// We check if the expected group was added to the menu
   $this->assertCount(1, $groups);
   $this-assertEquals('FooBar', $groups[0]->getTitle());
}
```

Reply:
Thanks! Seems to work just fine.
I could almost swear I tried this earlier and it didn't work